packages/react-components/src/organisms/__tests__/EditModal.test.tsx

```diff
@@ -38,78 +77,144 @@ describe('when saving', () => {
+      expect(getByText(/^save/i).closest('button')).toBeDisabled();
+
+      unmount();
+      resolveSave();
+    });
+
+    it('prompts when trying to leave while saving', async () => {
+      const { getByText, getByTitle, unmount } = result;
+      userEvent.click(getByText(/^save/i));
+
+      userEvent.click(getByTitle(/close/i));
+      expect(getUserConfirmation).toHaveBeenCalled();
+
+      unmount();
+      resolveSave();
+    });
+
     describe('and the save succeeds', () => {
-      let resolveSave!: () => void;
-      let result!: RenderResult;
       beforeEach(() => {
-        const handleSave = () =>
-          new Promise<void>((resolve) => {
-            resolveSave = resolve;
-          });
-        result = render(
-          <EditModal title="Title" backHref="#" onSave={handleSave}>
-            {() => null}
-          </EditModal>,
-        );
+        act(resolveSave);
       });
 
-      it('disables the save button while saving', async () => {
-        const { getByText, unmount } = result;
+      it('navigates to the back href without prompting', async () => {
+        const { getByText } = result;
 
         userEvent.click(getByText(/^save/i));
-        expect(getByText(/^save/i).closest('button')).toBeDisabled();
-
-        unmount();
-        act(resolveSave);
+        await waitFor(() => {
+          expect(history.location.pathname).toBe('/back');
+        });
+        expect(getUserConfirmation).not.toHaveBeenCalled();
       });
 
-      it('re-enables the save button after saving', async () => {
+      it('re-enables the save button', async () => {
         const { getByText } = result;
 
         userEvent.click(getByText(/^save/i));
-        act(resolveSave);
-
         await waitFor(() =>
           expect(getByText(/^save/i).closest('button')).toBeEnabled(),
         );
       });
+
+      // hypothetical cases if navigating back has not already caused unmount
+      describe('and the modal is still open', () => {
+        beforeEach(async () => {
+          const { getByText, rerender } = result;
+
+          userEvent.click(getByText(/^save/i));
+          await waitFor(() =>
+            expect(getByText(/^save/i).closest('button')).toBeEnabled(),
+          );
+
+          // not going to be dirty anymore since the values have just been saved
+          rerender(
+            <Router history={history}>
+              <EditModal {...props} backHref="/back" onSave={handleSave} />
+            </Router>,
+          );
+        });
+
+        it('does not prompt when trying to leave', async () => {
+          const { getByTitle, getByText } = result;
+          userEvent.click(getByTitle(/close/i));
+          expect(getUserConfirmation).not.toHaveBeenCalled();
+
+          await waitFor(() =>
+            expect(getByText(/^save/i).closest('button')).toBeEnabled(),
+          );
+        });
+
+        it('prompts when trying to leave after making edits again', async () => {
```

Comment:
This be deeeeeeeeeep...as in user action depth.

* modal initially created
* user changes things
* user saves
* modal still open (can't guarantee in this component that saving will always unmount the modal)
* user changes things again
* user closes
* expectation: gets a prompt

Crucially, when the modal becomes pristine (`!dirty`) again, it needs to reset from `hasSaved` to `initial` so that when it then becomes `dirty` again it'll prompt...